lfucg-exactions/server/accounts/viewsets.py

```diff
@@ -270,38 +270,52 @@ def create(self, request):
 
         data_set['created_by'] = self.request.user.id
         data_set['modified_by'] = self.request.user.id
-
-        if 'lot' in self.request.data:
-            serializer = AccountLedgerSerializer(data=data_set)
-            if serializer.is_valid(raise_exception=True):
-                self.perform_create(serializer)
-                return Response(serializer.data)
-
-        elif 'plat' in self.request.data:
-            chosen_plat = self.request.data['plat']
-            plat_set = Plat.objects.filter(id=chosen_plat)
-            non_sewer_credits_per_lot = 0
-            sewer_credits_per_lot = 0
-
-            if plat_set.exists():
-                buildable_lots = plat_set[0].buildable_lots
-
-                try:
-                    non_sewer_credits_per_lot = round((int(data_set['non_sewer_credits']) / buildable_lots), 2)
-                    sewer_credits_per_lot = round((int(data_set['sewer_credits']) / buildable_lots), 2)
-                except Exception as exc:
-                    return Response('Invalid credit entry', status=status.HTTP_400_BAD_REQUEST)
-
-            chosen_lots = Lot.objects.filter(plat=chosen_plat)
-            for lot in chosen_lots:
-                data_set['lot'] = lot.id
-                data_set['non_sewer_credits'] = non_sewer_credits_per_lot
-                data_set['sewer_credits'] = sewer_credits_per_lot
-
+        if data_set['entry_type'] == 'USE':
+            if data_set['lot']:
                 serializer = AccountLedgerSerializer(data=data_set)
                 if serializer.is_valid(raise_exception=True):
                     self.perform_create(serializer)
-            return Response('Success')
+                    return Response(serializer.data)
+
+            else:
+                chosen_plat = self.request.data['plat']
+                plat_set = Plat.objects.filter(id=chosen_plat)
+                non_sewer_credits_per_lot = 0
+                sewer_credits_per_lot = 0
+
+                if plat_set.exists():
+                    buildable_lots = plat_set[0].buildable_lots
+
+                    try:
+                        non_sewer_credits_per_lot = round((int(data_set['non_sewer_credits']) / buildable_lots), 2)
+                        sewer_credits_per_lot = round((int(data_set['sewer_credits']) / buildable_lots), 2)
+                    except Exception as exc:
+                        return Response('Invalid credit entry', status=status.HTTP_400_BAD_REQUEST)
+
+                chosen_lots = Lot.objects.filter(plat=chosen_plat)
+                for lot in chosen_lots:
+                    lot_serialized = LotSerializer(lot).data['lot_exactions']
+                    sewer_cap_percent = round(float(lot_serialized['dues_sewer_cap_dev'] / lot_serialized['sewer_due']), 2)
+                    sewer_trans_percent = round(float(lot_serialized['dues_sewer_trans_dev'] / lot_serialized['sewer_due']), 2)
```

Comment:
@Wei-KeatKong - is this where the float should come before the division? @KellyInnovation and I were talking about this this morning...

Reply:
do the float before the division